src/rest_vol_dataset.c

```diff
@@ -4137,4 +4164,165 @@ rv_dataset_write_cb(hid_t mem_type_id, hid_t mem_space_id, void *buf, struct res
+
+        /* For contiguous, the stride should be 1. */
+        for (i = 0; i < ndims; i++) {
+            if (stride[i] > 1)
+                FUNC_GOTO_DONE(FALSE);
+        }
+
+        if (nblocks > 1) {
+            /* Multiple blocks: count should be 1 except for the last dimension (fastest) */
+            for (i = 0; i < ndims - 1; i++) {
+                if (count[i] > 1)
+                    FUNC_GOTO_DONE(FALSE);
+            }
+        }
+
+        /* For contiguous, all faster running dimensions than the current dimension should be selected
+         * completely */
+        whole = (start[ndims - 1] == 0) && (count[ndims - 1] * block[ndims - 1] == dims[ndims - 1]);
+        for (i = ndims - 2; i >= 0; i--) {
+            if ((dims[i] > 1) && (count[i] * block[i] > 1) && !whole)
+                FUNC_GOTO_DONE(FALSE);
+
+            whole = whole && (start[i] == 0) && (count[i] * block[i] == dims[i]);
+        }
+    } /* end if*/
+    else if (H5S_SEL_POINTS == H5Sget_select_type(space_id)) {
+        /* Assumption: any point selection is non-contiguous in memory */
+        FUNC_GOTO_DONE(FALSE);
+    } /* end else if */
+
+done:
+    if (block)
+        RV_free(block);
+    if (count)
+        RV_free(count);
+    if (dims)
+        RV_free(dims);
+    if (stride)
+        RV_free(stride);
+    if (start)
+        RV_free(start);
+
+    return ret_value;
+} /* end RV_dataspace_selection_is_contiguous() */
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_convert_start_to_offset
+ *
+ * Purpose:     Convert starting position value to an offset value.
+ *
+ * Return:      Offset value on success/Negative value on failure.
+ *
+ * Programmer:  Jan-Willem Blokland
+ *              August, 2023
+ */
+static hssize_t
+RV_convert_start_to_offset(hid_t space_id)
+{
+    hsize_t *dims      = NULL;
+    hsize_t *start     = NULL;
+    hssize_t ret_value = 0;
+    int      ndims, i;
+
+    if (H5S_SEL_HYPERSLABS == H5Sget_select_type(space_id)) {
+        if ((ndims = H5Sget_simple_extent_ndims(space_id)) < 0)
+            FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, -1, "can't retrieve dataspace dimensionality");
+
+        if (NULL == (dims = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*dims))))
+            FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, -1,
+                            "can't allocate space for dimension 'dims' values");
+
+        if (H5Sget_simple_extent_dims(space_id, dims, NULL) < 0)
+            FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTGET, -1, "can't get dataspace dimension size");
+
+        if (NULL == (start = (hsize_t *)RV_malloc((size_t)ndims * sizeof(*start))))
+            FUNC_GOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, -1,
+                            "can't allocate space for hyperslab selection 'start' values");
+
+        if (H5Sget_regular_hyperslab(space_id, start, NULL, NULL, NULL) < 0)
```

Comment:
Same comment here about checking for a regular hyperslab with `H5Sis_regular_hyperslab`. Alternatively, I think that `H5Sget_select_bounds` may still fit this use case and should work with both regular and non-regular hyperslabs so that you don't have to worry about how to handle non-regular hyperslabs.

Reply:
This has been implemented.